mpmp/utilities/analysis_utilities.py

```diff
@@ -109,33 +138,38 @@ def load_purity_by_cancer_type(results_dir, sample_info_df):
     ---------
     results_dir (str): directory containing results files
     sample_info_df (pd.DataFrame): contains cancer type info for samples
+    classify (bool): look for classification results if true
 
     Returns
     -------
-    results_df (pd.DataFrame): results of classification experiments
+    results_df (pd.DataFrame): results of prediction experiments
     """
     results_df = pd.DataFrame()
     results_dir = Path(results_dir)
     for results_file in results_dir.iterdir():
         if not results_file.is_file(): continue
         results_filename = str(results_file.stem)
-        if 'preds' not in results_filename: continue
+        if classify and ('classify' not in results_filename
+                         or 'preds' not in results_filename): continue
+        if not classify and ('regress' not in results_filename
+                             or 'preds' not in results_filename): continue
         if results_filename[0] == '.': continue
         if check_compressed_file(results_filename):
-            training_data = '_'.join(results_filename.split('_')[:-4])
+            training_data = '_'.join(results_filename.split('_')[:-5])
```

Comment:
(Minor) Not sure what's the template of the file name, but in the future, it might be better to use a regular expression here to avoid any issues with a wrong negative index.

Reply:
Yeah, good point - this filename parsing code is kind of a mess at the moment. Part of the problem is that some of the training datatype identifiers have underscores in them ("me_27k", "me_450k") and some don't ("expression"), and I'm also using underscores to separate parts of the filename.

Regex could be a solution, but I often find them harder to interpret and debug than just using basic Python string operations, so I try to do the latter whenever I can.

I'll create an issue to think about the filenames in the future.